gitlab/v4/objects/files.py

```diff
@@ -238,7 +239,7 @@ def raw(
         Returns:
             The file content
         """
-        file_path = file_path.replace("/", "%2F").replace(".", "%2E")
+        file_path = utils.clean_str_id(file_path)
```

Comment:
Do we know for sure we don't need `%2E` from `.` in _all_ cases?

I guess this is similar to https://github.com/python-gitlab/python-gitlab/pull/1766 and the behavior was corrected in upstream gitlab. But I would feel a bit safer if we had a functional test for this to ensure no regressions. Maybe not for submodules below as that's a hassle, but for files, WDYT?

Reply:
I will work on adding a functional test to test this.

Reply:
So we are already testing a file with a period '.' here: https://github.com/python-gitlab/python-gitlab/blob/main/tests/functional/api/test_repository.py#L31

Reply:
Added a test for `raw()` using the filename `README.rst`. There are already tests for `create()`, `update()`, and `blame()` that use files with periods in them.